### ISSUE TITLE
UP006 does not report type annotations in quotes

### ISSUE THREAD

**Issue by roikoren755** [charliermarsh/ruff#3555]
First of all, thank you so much for a great linter! We recently moved from `flake8` to `ruff`, which massively sped up our linting step, and allowed us to check for things we couldn't previously use `flake8` plugins for, like `flake8-type-checking`, due to having to support Python 3.6. This just worked with `ruff`, without any hassle! 👏 

We recently updated one of our repositories to a newer version of Python, and wanted to turn on UP006, specifically, when we ran into what we believe is a bug.
If the typing annotation that includes a generic from typing that is no longer necessary, according to PEP585, the UP006 rule does not flag it as an error.

```python
from typing import Dict

does_not_flag: "Dict[str, str]" = {}
does_flag: Dict[str, str] = {}
```

The command used to run is `ruff check up006.py --isolated --select UP006`, and the output of `ruff --version` is `ruff 0.0.256`.

Once again, thanks for all your work!

**Comment by charliermarsh** [comment]
Thanks for the kind words :)

I _think_ we can check and fix these but I need to do some research -- there are some unintuitive corner cases around quoted annotations. We should almost certainly be flagging them as errors though, even if we can't autofix them.

(Related to #3508.)

**Comment by charliermarsh** [comment]
For now, made sure that we flag these. In the future, we'll be able to autofix _some_ cases, but not all (e.g., this is valid, but hard to fix: `x: "Li" "st[int] = []`).
